torchaudio/csrc/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+get_property(TORCHAUDIO_THIRD_PARTIES GLOBAL PROPERTY TORCHAUDIO_THIRD_PARTIES)
+
+################################################################################
+# Stuff common to libtorchaudio and _torchaudio.so
+################################################################################
+set(
+  LIBTORCHAUDIO_SOURCES
+  sox/io.cpp
+  sox/utils.cpp
+  sox/effects.cpp
+  sox/effects_chain.cpp
+  )
+
+if(BUILD_TRANSDUCER)
+  list(APPEND LIBTORCHAUDIO_SOURCES transducer.cpp)
+endif()
+
+################################################################################
+# libtorchaudio.so
+################################################################################
+if(BUILD_LIBTORCHAUDIO)
+  add_library(
+    libtorchaudio
+    SHARED
+    ${LIBTORCHAUDIO_SOURCES}
+    )
+  set_target_properties(libtorchaudio PROPERTIES PREFIX "")
+
+  target_include_directories(
+    libtorchaudio
+    PRIVATE
+    ${CMAKE_CURRENT_PROJECT_DIR}
+    )
+
+  target_link_libraries(
+    libtorchaudio
+    ${TORCHAUDIO_THIRD_PARTIES}
+    ${PROJECT_SOURCE_DIR}
+    )
+
+  install(
+    TARGETS
+    libtorchaudio
+    )
+
+  set(TORCHAUDIO_LIBRARY -Wl,--no-as-needed libtorchaudio -Wl,--as-needed CACHE INTERNAL "")
+endif()
+
+################################################################################
+# _torchaudio.so
+################################################################################
+if (BUILD_PYTHON_EXTENSION)
```

Comment:
suggestion: I'd go with BUILD_TORCHAUDIO_PYTHON_EXTENSION just because this is quite generic